Frontend + Backend Integration

Amaan17 opened the pull request:
# Task
#31 #29 

# Root Cause
The develop branch did not contain any code from the vector database.

# Solution 
Vector database code has been successfully integrated with the rest of the code. Note this does not mean communication between the different features but rather all the components working independantly in one code base.

# How Has This Been Tested? 
Manual testing, the following functions were tested:
- Front end displayed successfully.
- Transcription service working
- AI (online mode) responded correctly.
- Vector database was created and populated.

# Checklist: 
- [ ✅] I have linked issue to this Pull Request 
- [✅ ] I have performed a self-review of my code 
- [✅ ] I have commented my code, particularly in hard-to-understand areas 
- [ NA ] I have made corresponding changes to the documentation 
- [ ✅] I have added tests that prove my fix is effective or that my feature works 
- [ Manual testing] New and existing unit tests pass locally with my changes

Commit:
Uploaded homeplage template code

Commit:
Made only the AI assistant chat scrollable

Commit:
Fixed alignment for new project modal

Commit:
Removed footer

Commit:
Merge branch 'develop' of https://github.com/Monash-FIT3170/2025W2-QualAI into feat/EmilySalievski/#3-homepage-template

Commit:
Updated to use tailwind

raj0s committed:
Add functionality for transcribing
-Transcribe mp3 files to text printed to command line

raj0s committed:
Make changes to the formatting
-Adding a welcome page
-Rewriting the transcript function

Commit:
Removed default css to customise upload audio button

Commit:
Removed default css in index

RackyDay4 committed:
Implement file upload for backend
Implement ability to transcribe uploaded files
Fixed transcription function

raj0s committed:
Fix exception, to create directory instead of throwing error message
- Add Interview Uploads in git ignore to have a unique interview upload per machine

EamonSom committed:
Fix minor bug with transcription upload folder

RackyDay4 committed:
Refactor transriber class so interview is only passed in when actually trascribing it

Mohidkz committed:
added ollama to the docker container

Mohidkz committed:
connected backend and front end

- embedded the ollama model within the docker container
-handled the port within the backend python code
-used fastapi as a bridge between ollama and the frontend

lisag678 committed:
feat(ai chatbot): added scroll bar

lisag678 committed:
feat(AI Assistant): test different deepseek model

Mohidkz committed:
feat(ai chat): added implementation for ai communication. This change was made to fix the errors with said feature.

lisag678 committed:
feat(AI chatbot): remove AI response <think> text

This change is made to ensure that the AI output is cleaner without its 'thinking' text

achal1739 committed:
added requirements for modules needed in vector db

achal1739 committed:
Created virtual environement in db folder

achal1739 committed:
ingest.py: loads source document, splits text into chunks, prepares embedding model, push to db

achal1739 committed:
Added milestone 1 spec pdf for mock data to be populated within db

achal1739 committed:
Added querying functionality to the vector db

tommykpp committed:
feat(backend): implement functionality for
downloading a text file of transcription

achal1739 committed:
changed python version in backend docker file and added COPY commands

achal1739 committed:
Added new modules to requirements

achal1739 committed:
Implemented fast api within backend to have unique routes e.g hello.py

achal1739 committed:
Reformatted data path for embedding model to fetch from

achal1739 committed:
Deleted redundant vector_db folder used for testing setup

EamonSom committed:
feat(transcription): edit transcription functions
- Add the capacity to edit transcriptions in a text box before downloading
- Minor reconfigure of downloading functionaltiy to allow for editing transcription to be saved

EamonSom committed:
bug(transcription) minor bug fix on download
- Bug identified where files with spaces would break downloads if multiple files had already been downloaded.
- Add fix where spaces are replaced with '_' in file name.

achal1739 committed:
dockerized the ingest.py run command into the backend docker file so executes on compose up

lisag678 committed:
feat(ai chatbot): format chatbot md responses

This change is made so that the chatbot responses, which are in md, are formatted correctly to the user.

lisag678 committed:
feat(ai chatbot): added more chatbot formatting

This change is made to add bootstrap logos to the chatbot and remove some existing placeholder text.

EamonSom committed:
Docs(transcription) Added function documentation

guiltyeaten2 committed:
Merge remote-tracking branch 'origin/feat/mohid/#4-ollama-backend-connection' into feat/Joel/#18-integration

guiltyeaten2 committed:
integration(ai bot) merged ai bot branch to develop

guiltyeaten2 committed:
Merge remote-tracking branch 'origin/feat/Aryaman-R/#2-transcription-uploading' into feat/Joel/#18-integration

guiltyeaten2 committed:
integration(transcription) merged the backend of transcription to develop

guiltyeaten2 committed:
Integration(transcription gen) implement the transcription logic into the current frontend

guiltyeaten2 committed:
Merge remote-tracking branch 'origin/feat/Achal/#3-sentence-transformers' into feat/Joel/#18-integration

guiltyeaten2 committed:
Revert "Merge remote-tracking branch 'origin/feat/Achal/#3-sentence-transformers' into feat/Joel/#18-integration"

This reverts commit 2287c89e53307ea92bdc07ed155d37d6b809332b, reversing
changes made to d442fcb265e6e65aaa8171fe7b67fc84218e60da.

guiltyeaten2 committed:
reverted a commit and integrated the download function

Gunnraj-D committed:
FIX: fixed download transcription button on front end

Gunnraj-D committed:
FIX: added capacity to edit transcription and download edited transcription

Gunnraj-D committed:
TEMP FIX: upload transcript button now shows processing

Mohidkz committed:
tried fixing the compose error

Discomelons committed:
Reconfigurated the docker file by adding pip download. Additionally, SpaCy was implemented in the Vector DB to ensure retrieval makes better sense.

lisag678 committed:
feat(chatbot): Added online version of chatbot (google gemini API)

This change is made for the purpose of adding an online chatbot version for users (faster).

lisag678 committed:
feat(chatbot): Formatted online/offline button

Mohidkz05 committed:
Merge branch 'develop' into feat/lisa-mohid/#19-chatbotfix

Mohidkz05 committed:
fixed the duplication errors made during merging

Discomelons committed:
Merge remote-tracking branch 'origin/develop' into Amaan-Achal-Farhan-Backend-Optimisation-And-spaCy-#31. Code merge with original develop - pre-AI

Discomelons committed:
Adjusted files to ensure the vector database functions correctly and can exist with the frontend code.

Amaan17 committed:
Merge remote-tracking branch 'origin/feat/lisa-mohid/#19-chatbotfix' into Amaan-Achal-Farhan-Backend-Optimisation-And-spaCy-#31

Amaan17 committed:
Finalised Integration. Chatbot, front-end and vector database are now situated in a single project.

Amaan17 committed:
Adjusted docker file to include database path and install ffmpeg correctly.

Mohidkz05 reviewed (approved):
Amaan demonstrated its functionality and based off the files the integration seems to be fine

Amaan17 committed:
Merge branch 'develop' into Amaan-Achal-Farhan-Backend-Optimisation-And-spaCy-#31

Amaan17 committed:
Fix - Vector Database and Ingestion

Removed the shell script and replaced it with the python one to ensure Vector Databases creation and ingestion.

Gunnraj-D committed:
FIX: AI assistant no longer has double text and no longer uses uninitialised variable

Gunnraj-D committed:
FIX: added development methof for quicker docker build

FarhanM3135 committed:
Bug fixed:
Vector database looking in hte wrong directory
